### ISSUE TITLE
API Requirements as discussed with Alexion

### ISSUE THREAD

**Issue by knoddin** [monarch-initiative/phenotypr#1]
Below is some information re: our (Kara and my) meeting with Alexion.

Kara showed them this mockup for the front-end question flow:

[Link to mockup](https://projects.invisionapp.com/share/RZEABQ2MQ#/screens/274255502_Body-Landing) : just work your way through the set of questions and you can see the basic structure – everything is pre-filled/answered.

Alexion has decided that a JAVA servlet structure, rather than a traditional API, would be best given their code and algorithms. I was told that the servlet would still have HTTP protocol and function like a RESTful API.

We decided that the first set of questions would be static (all answers would POST/be saved to whatever back end we have):

                -name or id
                -answering for self or someone else
                -patient’s age
                -patient’s sex

The field where symptoms are entered is where the algorithm and specialized set of questions begins. Symptoms are posted --> user gets sent back first question in the algorithm --> user posts response --> user gets next question, etc. until the output (HPO lay terms) is generated.

Alexion is also going to figure out how to incorporate sessions into their code to make sure that the algorithm and survey is carried through and tied individually to each user. It is not currently set up to do that.

I’ve attached photos of the whiteboard that (somewhat) illustrate what we discussed (numbered in order 1-3): [API_meeting_w%2f_Alexion.zip](https://github.com/monarch-initiative/phenotypr/files/1776560/API_meeting_w.2f_Alexion.zip)

**Comment by jmcmurry** [comment]
@pnrobinson and @julesjacobsen could you please weigh in? Thanks

**Comment by pnrobinson** [comment]
I am beginning to think we need to put a lot more effort into this topic, and that we need to develop out a medical questionaire ontology to guide software to ask the right questions. The whiteboard stuff was not illuminating in terms of the overall approach....

**Comment by jmcmurry** [comment]
Agreed, but we need to be done by June. What is feasible before then?

**Comment by mellybelly** [comment]
I still think that simple questions about body parts with a visual, that then allows autocomplete for each body part would be a) simple to implement with subgraphs of HPO, b) get us our full profiles - which is the eval we are doing for PCORI. We could also play around more with export tools such as phenopackets. Seems to me that the what questions to ask next for the differential are needed much more by the clinician, at least at this time.  @kshefchek

**Comment by jmcmurry** [comment]
Agreed, but this ticket is *just for the API part.

**Comment by julesjacobsen** [comment]
I though that, given the time and resources we have available, we'd all agreed that we were going with the far simpler systems based approach using autocomplete and the lay-persons HPO? We already/almost have an API set up for this (is this right @kshefchek ?) and the UI team need to be working against this.

**Comment by camenden** [comment]
Does this mean the UI is ready for us to work against? Or not yet?

Thanks,
Catherine

Sent from my iPhone

On Mar 6, 2018, at 4:40 AM, Jules Jacobsen <notifications@github.com<mailto:notifications@github.com>> wrote:


I though that, given the time and resources we have available, we'd all agreed that we were going with the far simpler systems based approach using autocomplete and the lay-persons HPO? We already/almost have an API set up for this (is this right @kshefchek<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_kshefchek&d=DwMFaQ&c=qS4goWBT7poplM69zy_3xhKwEW14JZMSdioCoppxeFU&r=kQEw5tvo_o0yyaTgLiWUNNfPeMx-GYcAPHD94xp1uNy0RWLHNRp31FMaw2VL4kqI&m=HZp1ChUmrBjgDD0OWUXejUcF23XLfmJAS9wrWKtlWys&s=uxmdaVDUKVnxuCRVGhxhLkir-edYiZtUF_8-IYXdsMo&e=> ?) and the UI team need to be working against this.

-
